File: examples/OmeroClients/sudo.m
client = omero.client();
sudoClient = omero.client();

try
    sf = client.createSession('root','ome');
    sessionSvc = sf.getSessionService();

    p = omero.sys.Principal();
    p.name = 'root'; % Can change to any user
    p.group = 'user';
    p.eventType = 'User';
    
    sudoSession = sessionSvc.createSessionWithTimeout( p, 3*60*1000 ); % 3 minutes to live
    
    sudoSf = sudoClient.joinSession( sudoSession.getUuid().getValue() );
    sudoAdminSvc = sudoSf.getAdminService();
    disp(sudoAdmin.Svc.getEventContext().userName);
    
catch ME
    
    sudoClient.closeSession();
    client.closeSession();

end

File: examples/OmeroClients/sudo.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import sys
import omero

args = list(sys.argv)
client = omero.client(args)
sudoClient = omero.client(args)
try:
    sf = client.createSession("root", "ome")
    sessionSvc = sf.getSessionService()

    p = omero.sys.Principal()
    p.name = "root"  # Can change to any user
    p.group = "user"
    p.eventType = "User"

    # 3 minutes to live
    sudoSession = sessionSvc.createSessionWithTimeout(p, 3*60*1000L)

    sudoSf = sudoClient.joinSession(sudoSession.getUuid().getValue())
    sudoAdminSvc = sudoSf.getAdminService()
    print sudoAdminSvc.getEventContext().userName

finally:
    sudoClient.closeSession()
    client.closeSession()
